### PR TITLE
 Add stdlib.{arch,license,pushd,pushenv}.

### PR DESCRIPTION
**Pull request by Arignir**
This pull request add some type and functions that will be used throughout the compilation library.

### CODE REVIEW

File: stdlib/license.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3.6
+# -*- coding: utf-8 -*-
+"""An enumeration of possible a softwares' licenses."""
```
Comment (Pedrochu) [inline stdlib/license.py]: Should be `of possible softwares' licenses`.

**Review by Pedrochu — CHANGES_REQUESTED** [review]

File: stdlib/license.py (hunk 1 above)
Comment (doom) [reply stdlib/license.py]: `Software` is uncountable, so more like `of possible software licenses`

File: stdlib/license.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+from enum import Enum
+
+
+class License(Enum):
+    """An enumeration of a softwares' licenses.
```
Comment (doom) [inline stdlib/license.py]: Same here

File: stdlib/pushd.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env python3.6
+# -*- coding: utf-8 -*-
+"""Provide a function to save the current working directory and switch to the given one for the duration of a new context."""
```
Comment (doom) [inline stdlib/pushd.py]: `Provides`

File: stdlib/pushenv.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env python3.6
+# -*- coding: utf-8 -*-
+"""Provide a function to save the current environment for the duration of a new context."""
```
Comment (doom) [inline stdlib/pushenv.py]: `Provides`

**Review by doom — APPROVED** [review]

**Pull request merged**